helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -138,15 +134,16 @@ void release(AssignableReplica assignableReplica) throws IllegalArgumentExceptio
           getInstanceName());
       return;
     }
-    Set<String> partitions = _currentAssignments.get(resourceName);
-    if (!partitions.contains(partitionName)) {
+    Map<String, String> partitionStateMap = _currentAssignments.get(resourceName);
```

Comment:
Probably a good idea to add a comment here what is mapping to what.

Reply:
partitionStateMap => <partition, state> map?
I think it is clear enough.